pkg/datasource.go

```diff
@@ -162,29 +170,74 @@ func (o *OCIDatasource) QueryData(ctx context.Context, req *backend.QueryDataReq
 }
 
 func (o *OCIDatasource) testResponse(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
-	return &backend.QueryDataResponse{}, nil
-	// var ts GrafanaCommonRequest
-
-	// query := req.Queries[0]
-	// if err := json.Unmarshal(query.JSON, &ts); err != nil {
-	// 	return &backend.QueryDataResponse{}, err
-	// }
-
-	// //o.logger.Error("ts.Com is " + ts.Compartment)
-	// listMetrics := monitoring.ListMetricsRequest{
-	// 	CompartmentId: common.String(ts.Compartment),
-	// }
-	// reg := common.StringToRegion(ts.Region)
-	// o.metricsClient.SetRegion(string(reg))
-	// res, err := o.metricsClient.ListMetrics(ctx, listMetrics)
-	// if err != nil {
-	// 	return &backend.QueryDataResponse{}, err
-	// }
-	// status := res.RawResponse.StatusCode
-	// if status >= 200 && status < 300 {
-	// 	return &backend.QueryDataResponse{}, nil
-	// }
-	// return nil, errors.Wrap(err, fmt.Sprintf("list metrircs failed %s %d", spew.Sdump(res), status))
+
+	// GrafanaCommonRequest - captures the common parts of the search and metricsRequests
+	// the structure is used here to retrieve the TenancyOCID and CompartmentOCID
+	var ts GrafanaCommonRequest
+	query0 := req.Queries[0]
+
+	if err := json.Unmarshal(query0.JSON, &ts); err != nil {
+		return &backend.QueryDataResponse{}, err
+	}
+	o.logger.Debug("Testing OCI logs datasource", "TenancyOCID/CompartmentOCID", ts.TenancyOCID+"/"+ts.Compartment)
+
+	listLogsGroup := logging.ListLogGroupsRequest{
```

Comment:
I do not think I understood this point. In order to test if a log stream is accessible I need to reference the log group it belongs. Otherwise the query will not work. Maybe I am missing something here.

Reply:
What I mean is that the log group name/ID and the log object name/ID fields are optional in a logging search query. A query such as the following is perfectly valid and will search all log groups and log objects within the specified compartment:

`search "ocid1.compartment.oc1..<rest of compartment OCID" | where substr(data.namespace, 0, 5) = 'cne-i' | summarize count data.podname by data.namespace`

So there really is really no need in this logic to specify a log group or log object.

Reply:
Even if I am able to use this type of query in console (when logged in as tenancy admin) I am not able to use this query in the test function nor in any dashboard in Grafana. In both case I got an Autorization error:
```
logger=plugin.oci-logs-datasource t=2022-08-24T06:52:00.9+0000 lvl=dbug msg="Logging query will NOT return numeric data" refId=A
logger=plugin.oci-logs-datasource t=2022-08-24T06:52:00.9+0000 lvl=dbug msg="Processing log records search query" panelId=2 query="search \"ocid1.compartment.oc1..xxxxxxx\"" refId=A to=2022-08-24T06:37:03.586Z from=2022-08-24T00:37:03.586Z
```

I suspect it may be a missing policy in  my Dynamic Group, but I have already the read log and log groups for the tenancy already assigned in my Dynamic group, so I have no idea what else should I put. My concern is that this kind of test may fail in most cases even if the policy is sufficient to retrieve the logs, it may not be sufficient to pass this test.

Reply:
Interesting since I never include the log group or lob object name/IDs in the queries I use in the logs data panels. So it must be a permissions issue. You have the following permissions associated with your dynamic group?
```
allow group GrafanaLoggingUserGroup to read log-groups in tenancy
allow group GrafanaLoggingUserGroup to read log-content in tenancy
allow group GrafanaLoggingUserGroup to read compartments in tenancy
```

Reply:
yes, I do have, but I am using dynamic group and not User group:

```
allow dynamicgroup arm_dynamic_group to read compartments in tenancy
allow dynamicgroup arm_dynamic_group to read log-groups in tenancy
allow dynamicgroup arm_dynamic_group to read log-content in tenancy
```